doc/upgrade.md

```diff
@@ -0,0 +1,136 @@
+
+## Problem 1: device not found in devicegraph
+
+As mentioned above, the device name of an encryption device could not be found in the probed devicegraph. The reason is because storage-ng probes the encryption devices with an auto-generated name (e.g., "cr-auto-1"), and in general, that auto-generated name does not match with the name from the fstab entry.
+
+So, if the fstab has an entry like the following:
+
+```
+/dev/mapper/cr_ata-111-part2    /home   ext4    defaults 0 0
+```
+
+that "/dev/mapper/cr_ata-111-part2" will not be found in the devicegraph because there is nothing probed with that name. To avoid this problem, the crypttab file of the Operating System to upgrade is parsed in addition to its fstab file.
+
+Storage-ng generates the crypttab entries taking into account the default `mount_by` option. So, the crypttab file could have entries like:
+
+```
+cr_home     UUID=000-111-222                            # default mount_by = uuid
+cr_home     /dev/sda2                                   # default mount_by = device name
+cr_home     /dev/sda2                                   # default mount_by = volume label
+cr_home     /dev/disk/by-id/scsi-11212-part2            # default mount_by = device id
+cr_home     /dev/disk/by-path/pci-0002-part2            # default mount_by = device path
+cr_home     /dev/disk/by-uuid/000-111-222
+```
+
+Several things to take into account here. When the UUID= format is used, that UUID refers to the LUKS device instead of the the underlying device. When the default mount_by is "volume label", the kernel name of the underlying device is used. When using "device id" or "device path" as default mount_by, the proper udev name of the underlying device is used. And finally, the user could indicate a device by using uuid udev name.
+
+Once the crypttab file is parsed, the information about encryption names is stored in the devicegraph to be able to find an encryption device when it was probed with another name. For example, if we have a fstab entry like:
+
+```
+/dev/mapper/cr_home     /home   ext4    defaults 0 0
+```
+
+and a crypttab entry like:
+
+```
+cr_home     /dev/sda2
+```
+
+the device "/dev/sda2" is searched in the devicegraph and the crypttab name "cr_home" is stored in its associated encryption device. So, in case that there is an encryption device over "/dev/sda2", for example "/dev/mapper/cr-auto-1", that encryption device will save "cr_auto" as alternative name.
+
+Once this information is saved in the encryption devices, when an encryption name is searched in the devicegraph, we can check if the searched named is the alternative name of any encryption device. So, for this example, when we search in the devicegraph the device by "/dev/mapper/cr_home", we can return "/dev/mapper/cr-auto-1".
+
+This problem was fixed for SLE-15, see [related PBI](https://trello.com/c/OGssTRzO/393-osdistribution-p2p1-1094963-opensuse-150-installation-or-upgrade-fails-if-disk-contains-encrypted-partition-luks).
```

Comment:
Not sure if I would link to Trello here. It's just a tool for team coordination (with restricted access). Better use something more stable (instead/in addition to) Trello, like the bugzilla entry or the pull request.